components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java

```diff
@@ -2913,6 +2918,15 @@ protected String addResourceFile(String resourcePath, ResourceFile resourceFile,
         private Registry registry;
         private boolean isTenantFlowStarted;
         private int tenantId;
+        private String userName;
+
+        public String getUserName() {
+            return userName;
+        }
+
+        public void setUserName(String userName) {
+            this.userName = userName;
+        }
```

Comment:
if this is only used in annonymous mode, shall we only keep a boolean - isAnnonomousMode?
Otherwise, having a username here might make confusions and might not used by people in the future as intended way.

Reply:
Fixed with e62ee98c11ed8cfe771cd0efa2aa1dbcfa7e7013